### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by mpetrun5**
**Short description of work done**

Implemented metrics endpoint and added example dashboard

### PR Checklist

- [x] I have run linter localy
- [x] I have run unit and integration tests locally
- [x] Rebased to master branch / merged master
- [x] Updated CHANGELOG.md

### Changes

- added /metrics endpoint
- embedded app version on build
- added prometheus and grafana to docker-compose 
- added grafana example dashboard

Open to suggestions:

Screenshot from 2020-12-03 11-22-47
Screenshot from 2020-12-03 11-22-31
Screenshot from 2020-12-03 11-22-12

### Issues

Closes: #137

### CODE REVIEW

**Review by MakMuftic — DISMISSED** [review]
From a code perspective, everything seems good.

I would maybe just paint _Failed request_ and _Penalized nodes_ to red

File: Makefile [hunk 1]
```diff
@@ -35,12 +33,16 @@ PLATFORMS := linux/amd64 windows/amd64 darwin/amd64 linux/arm
 temp = $(subst /, ,$@)
 os = $(word 1, $(temp))
 arch = $(word 2, $(temp))
+version = $(shell sed -n 's/version=//p' .version)
 
 $(PLATFORMS):
 	@if [ "$(os)" = "windows" ]; then \
-			GOOS=$(os) GOARCH=$(arch) go build -o 'build/windows/vedran.exe'; \
+			GOOS=$(os) GOARCH=$(arch) go build -ldflags "-X github.com/NodeFactoryIo/vedran/pkg/version.Version=$(version)" -o 'build/windows/vedran.exe'; \
```
Comment (mpetrunic) [inline Makefile]: can you define flag as variable instead of copy pasting everywhere?

**Review by mpetrunic — COMMENTED** [review]
Let's not merge this until we merge payout milestone on master.

Could you update Readme.md with instructions installing prometheus(link to official docs), installing grafana (official docs + adding prometheus data source) + dashboard import?

Also document where can grafa be found when ran trough docker-compose.

**Comment by mpetrunic** [comment]
LGTM, but draft until we merge payout to master

File: docker-compose.yml [hunk 2]
```diff
@@ -2,24 +2,58 @@ version: '3'
...
+      - 9090:9090
+
+  grafana:
+    image: grafana/grafana
+    volumes:
+      - grafana_data:/var/lib/grafana
+      - ./infra/grafana/provisioning:/etc/grafana/provisioning
+    environment:
+      - GF_SECURITY_ADMIN_USER=admin
+      - GF_SECURITY_ADMIN_PASSWORD=admin
```
Comment (mpetrunic) [inline docker-compose.yml]: you could remove auth alltogether. Something related to anonymus access

**Review by mpetrunic — COMMENTED** [review]
add comment to readme.md#demo with grafana url.

**Review by MakMuftic — APPROVED** [review]
LGTM

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**